lib/quantity.ex

```diff
@@ -68,21 +72,42 @@ defmodule Quantity do
   """
   @spec parse(String.t()) :: {:ok, t} | :error
   def parse(input) do
-    with [value_string, unit_string] <- String.split(input, " ", parts: 2),
+    with {:ok, value_string, unit_string} <- parse_split_value_and_unit(input),
          {:ok, value} <- Decimal.parse(value_string) do
-      unit =
-        cond do
-          unit_string =~ "/" -> [:div | String.split(unit_string, "/", parts: 2)] |> List.to_tuple()
-          unit_string =~ "*" -> [:mult | String.split(unit_string, "*", parts: 2)] |> List.to_tuple()
-          true -> unit_string
-        end
+      unit = parse_unit(unit_string)
 
       {:ok, new(value, unit)}
     else
       _ -> :error
     end
   end
 
+  defp parse_split_value_and_unit(input) do
+    case String.split(input, " ", parts: 2) do
+      [value] -> {:ok, value, "1"}
+      [value, unit] -> {:ok, value, unit}
+      _ -> :error
+    end
+  end
+
+  defp parse_unit(unit_string) do
+    if unit_string =~ "/" do
+      [:div | unit_string |> String.split("/", parts: 2) |> Enum.map(&parse_mult_unit/1)] |> List.to_tuple()
+    else
+      parse_mult_unit(unit_string)
+    end
+  end
+
+  defp parse_mult_unit(unit_string) do
+    unit_string
+    |> String.split("*")
+    |> Enum.map(&parse_base_unit/1)
+    |> Enum.reduce(&{:mult, &1, &2})
+  end
```

Comment:
Minor detail: I think this is nice, but also somewhat implicit, because it actually also parses non-mult units (because `acc` is returned immediately from `Enum.reduce` if the list has only one element).

Perhaps rename it to `parse_mult_or_single_unit`?

Reply:
I think, logically, it parses :mult units, which also includes the edge case of only one base unit. Since another function, `parse_base_unit/1`, handles the base case, I think `parse_mult_unit` is good.
Ok?